fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryFlatMapPlan.java

```diff
@@ -125,7 +127,7 @@ public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final
                             });
                 },
                 continuation,
-                5);
+                5).skipThenLimit(executeProperties.getSkip(), executeProperties.getReturnedRowLimit());
```

Comment:
At some point, we should also change this `5` constant to use the store's `PipelineSizer`, but that doesn't have to be now

Reply:
Agreed.